src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -0,0 +1,206 @@
+    // Defines Encoders 
+    Encoder liftEncoder = new Encoder(RobotMap.Encoders.LIFT_A, RobotMap.Encoders.LIFT_B);
+    // TODO we still need to calculate this
+    private final double encoderDistancePerTick = 0.01;
+    // Defines Motors 
+    WPI_TalonSRX liftMotorA = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_A);
+    WPI_TalonSRX liftMotorB = new WPI_TalonSRX(RobotMap.CAN.LIFT_MOTOR_B);
+    // Defines the previus motors as one motor 
+    SpeedControllerGroup liftDrive = new SpeedControllerGroup(liftMotorA, liftMotorB);
+    DoubleSolenoid liftTransmission = new DoubleSolenoid(RobotMap.Solenoids.LIFT_TRANSMISSION_A,
+            RobotMap.Solenoids.LIFT_TRANSMISSION_B);
+
+    final static double kP = Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1);
+    final static double kI = Preferences.getInstance().getDouble(RobotMap.Preferences.K_I, 1);
+    final static double kD = Preferences.getInstance().getDouble(RobotMap.Preferences.K_D, 1);
+    final static double kF = Preferences.getInstance().getDouble(RobotMap.Preferences.K_F, 1);
+
+    @Override
+    public void setAbsoluteTolerance(double t) {
+        super.setAbsoluteTolerance(0.05);
+    }
+
+    //enumerator that will be pulled from for the GoToHeight Command
+    public enum LiftHeight {
+        //will be changed
+        kFloor(0), kSwitch(1), kPortal(2), kIntake(3), kScaleLow(4), kScaleHigh(5), kClimb(6), kMaxHeight(7);
+
+        @SuppressWarnings("MemberName")
+        private double value;
+
+        //
+        LiftHeight(double value) {
+            this.value = value;
+        }
+
+        public double get() {
+            return value;
+        }
+    }
+
+    public Lift() {
+        super("Lift", kP, kI, kD, kF);
+        setOutputRange(Preferences.getInstance().getDouble(RobotMap.Preferences.DOWN_MAX_SPEED, 1),
+                Preferences.getInstance().getDouble(RobotMap.Preferences.UP_MAX_SPEED, 1));
+        //creates a child for the encoders
+        addChild(liftEncoder);
+        addChild(topLimitSwitch);
+        addChild(bottomLimitSwitch);
+
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_P, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_I, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_D, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.K_F, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.LIFT_UP_DOWN_INCREMENT, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.UP_MAX_SPEED, 1);
+        Preferences.getInstance().getDouble(RobotMap.Preferences.DOWN_MAX_SPEED, 1);
+    }
+
+    protected double returnPIDInput() {
+        return liftEncoder.getDistance();
+    }
+
+    protected void usePIDOutput(double output) {
+        if (topLimitSwitch.get() == true && output > 0) {
+            setSetpoint(LiftHeight.kMaxHeight.get());
```

Comment:
This should be using `setRelativeSetpoint(output)` instead

Reply:
You don't need the set relative here, but you DO NEED to add liftDrive.stop() so that we don't continue driving up. Then add the same thing for the bottom case.